pkg/volume/gce_pd/gce_util.go

```diff
@@ -58,6 +66,12 @@ func (util *GCEDiskUtil) AttachAndMountDisk(pd *gcePersistentDisk, globalPDPath
 	devicePath := ""
 	// Wait for the disk device to be created
 	for {
+		sdAfter, err := filepath.Glob("/dev/sd*")
+		if err != nil {
+			glog.Errorf("Error filepath.Glob(\"/dev/sd*\"): %v\r\n", err)
+		}
+		udevadmChangeToNewDrives(sdBefore, sdAfter)
+
```

Comment:
Actually @quinton-hoole's intuition is right here. This could cause kubelet's `syncPods` loop to sleep for an additional 10 seconds, see [kubelet.go#L1328](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubelet/kubelet.go#L1328), since volume teardown is synchronous (unlike [volume creation](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubelet/kubelet.go#L1166) which is asynchronous, see [pkg/kubelet/pod_workers.go#L151](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubelet/pod_workers.go#L151)). Any opposition to spawning of a new thread (go routine) to execute `vol.TearDown()` in `kubelet.go:cleanupOrphanedVolumes(...)`?